compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -1211,6 +1211,15 @@ impl<'a> Resolver<'a> {
         span: Span,
         condition: Expression,
     ) -> Option<ExprId> {
+        if let Some(
+            assert_message_expr @ Expression {
+                kind: ExpressionKind::Literal(Literal::Str(..)), ..
+            },
+        ) = assert_message_expr
+        {
+            return Some(self.resolve_expression(assert_message_expr));
+        }
```

Comment:
Can you move this into the check below so that the test for `Some(assert_message_expr @ ...` isn't duplicated?

Reply:
I've reworked this to avoid this duplication now.